kasa/tests/smart/modules/test_autooff.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+import sys
+from datetime import datetime
+from typing import Optional
+
+import pytest
+from pytest_mock import MockerFixture
+
+from kasa import Module
+from kasa.smart import SmartDevice
+from kasa.tests.device_fixtures import parametrize
+
+autooff = parametrize(
+    "has autooff", component_filter="auto_off", protocol_filter={"SMART"}
+)
+
+
+@autooff
+@pytest.mark.parametrize(
+    "feature, prop_name, type",
+    [
+        ("auto_off_enabled", "enabled", bool),
+        ("auto_off_minutes", "delay", int),
+        ("auto_off_at", "auto_off_at", Optional[datetime]),
+    ],
+)
+@pytest.mark.skipif(
+    sys.version_info < (3, 10),
+    reason="Subscripted generics cannot be used with class and instance checks",
+)
```

Comment:
What is causing this issue? Is it the instance check in the test or in the actual change in `autooff.py`?

Reply:
It's the instance check, older versions do not support `TypeUnion` (`X | None`) nor checking against `Optional`.